core/store/src/opener.rs

```diff
@@ -290,28 +314,34 @@ impl<'a> StoreOpener<'a> {
     fn ensure_kind(
         mode: Mode,
         opener: &DBOpener,
-        kind: DbKind,
-        which: &'static str,
+        archive: bool,
+        temp: Temperature,
     ) -> Result<(), StoreOpenerError> {
+        let which: &'static str = temp.into();
+        tracing::debug!(target: "db_opener", path=%opener.path.display(), archive = archive, which = which, "Ensure db kind is correct and set.");
```

Comment:
Note that `cargo fmt` doesn't handle complex macros like this, and manual formatting may be needed.

```suggestion
        tracing::debug!(target: "db_opener", path = %opener.path.display(), archive, which, "Ensure db kind is correct and set.");
```